app/controllers/users_controller.rb

```diff
@@ -1,3 +1,26 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find_by id: params[:id]
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+
+    if @user.save
+      flash[:success] = t "users.new.flash_success"
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit User::PERMITTED_PARAMS
```

Comment:
requrie có tác dụng gì? permit có tác dụng gì?

Reply:
- require(:param) đảm bảo param phải xuất hiện trong hash param, trả về lỗi nếu param đó không tồn tại. 
- permit trả lại hash param chỉ bao gồm các key được permit.

Reply:
In cái params.require(:user) vào đây em

Reply:
<ActionController::Parameters {"name"=>"Test User", "email"=>"test@gmail.com", "password"=>"123456", "password_confirmation"=>"123456"} permitted: false>

Reply:
![Screenshot from 2020-08-03 11-35-32](https://user-images.githubusercontent.com/68632706/89146309-7f348c80-d57d-11ea-9933-17fca3c686c7.png)